### ISSUE TITLE
queries_result_list bug in evaluation InformationRetrievalEvaluator.py

### ISSUE THREAD

**Issue by qixintechnology** [UKPLab/sentence-transformers#752]
Thanks for your work! I used bi_encoder to finetune information retrieval model. I encounter issue when apply evaluation class  InformationRetrievalEvaluator.

It looks there existing a bug when get queries_result_list. Should be [for query_itr in range(len(query_embeddings)):] -> [for query_itr in range(corpus_start_idx,corpus_end_idx):] to collect the correct queries result list.

Please kindly help to have a quick check on this. Thanks!

**Comment by nreimers** [comment]
From a first look I could not see a bug there. What do you think is broken and why?

**Comment by qixintechnology** [comment]
Thanks for your reply! Because [for query_itr in range(len(query_embeddings)):] is in the loop of [for corpus_start_idx in itr:]. query_itr will start from '0' each time, in this case, will generate repetitive value in queries_result_list.

**Comment by nreimers** [comment]
This is correct and intended. We split the corpus into chunks, eg. docs 0-100k and 100k-200k. We find from each chunk the top k docs and merge them in the results list.
